auth/presentation/src/main/java/com/gkp/auth/presentation/components/TaskyTextField.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.text.input.InputTransformation
+import androidx.compose.foundation.text.input.TextFieldLineLimits
+import androidx.compose.foundation.text.input.TextFieldState
+import androidx.compose.foundation.text.input.rememberTextFieldState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.gkp.core.designsystem.theme.TaskyGreen
+import com.gkp.core.designsystem.theme.TaskyLightGray
+import com.gkp.core.designsystem.theme.TaskyTextFieldColor
+import com.gkp.core.designsystem.theme.TaskyTextHintColor
+
+
+@Composable
+fun TaskyTextField(
+    modifier: Modifier = Modifier,
+    hintText: String,
+    textState: TextFieldState,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    inputTransformation: InputTransformation? = null,
+    endIcon: ImageVector? = null,
+    isEmailValid: Boolean = false,
+    endIconContentDescription: String? = null
+) {
+
+    val iconTintColor = if (isEmailValid) TaskyGreen else TaskyTextFieldColor
+    BasicTextField(
+        modifier = modifier.defaultMinSize(minHeight = 63.dp)
+            .background(
+                color = TaskyLightGray,
+                shape = RoundedCornerShape(10.dp)
+            )
+            .padding(8.dp),
+        state = textState,
+        textStyle = MaterialTheme.typography.bodyLarge.copy(
+            color = TaskyTextFieldColor
+        ),
+        keyboardOptions = keyboardOptions,
+        inputTransformation = inputTransformation,
+        lineLimits = TextFieldLineLimits.SingleLine,
+        decorator = { innerBox ->
+            Row(
+                modifier = modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                if (textState.text.isEmpty()) {
+                    Text(
+                        text = hintText,
+                        style = MaterialTheme.typography.bodyLarge.copy(
+                            color = TaskyTextHintColor
+                        )
+                    )
+                } else {
+                    innerBox()
+                }
```

Comment:
Your cursor is likely not showing because you don't draw the inner box if the text field is empty. The inner box contains the cursor drawing AFAIK

Reply:
Thanks the is solve by moving the else branch